packages/manager/apps/dedicated/client/app/dedicatedCloud/datacenter/backup/backup.service.js

```diff
@@ -0,0 +1,166 @@
+import find from 'lodash/find';
+import get from 'lodash/get';
+import map from 'lodash/map';
+import remove from 'lodash/remove';
+import sortBy from 'lodash/sortBy';
+
+import { BACKUP_OFFER_LEGACY, CATALOG_INFO } from './backup.constants';
+import Backup from './backup.class';
+import BackupOffer from './backup-offer.class';
+
+export default class BackupService {
+  /* @ngInject */
+  constructor($q, OvhApiDedicatedCloudDatacenter, OvhApiOrder, WucOrderCartService) {
+    this.$q = $q;
+    this.backupApi = OvhApiDedicatedCloudDatacenter.Backup().v6();
+    this.catalogApi = OvhApiOrder.CatalogFormatted().v6();
+    this.WucOrderCartService = WucOrderCartService;
+  }
+
+  addConfigToCart(cartItem, datacenterId, offerType) {
+    return this.$q
+      .all([
+        this.WucOrderCartService.addConfigurationItem(
+          cartItem.cartId,
+          cartItem.itemId,
+          'datacenter_id',
+          datacenterId,
+        ).$promise,
+        this.WucOrderCartService.addConfigurationItem(
+          cartItem.cartId,
+          cartItem.itemId,
+          'offer_type',
+          offerType,
+        ).$promise,
+      ])
+      .then(() => cartItem);
+  }
+
+  assignCart(cart) {
+    return this.WucOrderCartService.assignCart(cart.cartId);
```

Comment:
Why add another layer above `WucOrderCartService` ? I think it already provides the `/order/cart` api abstraction and it seems like there is no more specific logic added

Reply:
Its actually not required. But now the code returns the cart object again, to enabling chaining of calls